base_delivery_carrier_files/models/stock_picking.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2012 Camptocamp SA
+# Author: Guewen Baconnier
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def generate_carrier_files(self, auto=True, recreate=False):
+        """
```

Comment:
fails silently when a picking with  no carrier or no carrier.carrier_file_id  is selected.
Perhaps some feedback (like a warning icon), or  just filtering out non-eligible stock.picking from wizard could be a usability improvement.